index.js

```diff
@@ -0,0 +1,192 @@
+    return resArr;
+  }, {});
+};
+
+// Task 3
+/* Выведите массив ids для релизов у которых рейтинг 5.0. */
+
+const newReleases = [
+  {
+    id: 70111470,
+    title: "Die Hard",
+    boxart: "http://cdn-0.nflximg.com/images/2891/DieHard.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+    boxart: "http://cdn-0.nflximg.com/images/2891/BadBoys.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+  {
+    id: 65432445,
+    title: "The Chamber",
+    boxart: "http://cdn-0.nflximg.com/images/2891/TheChamber.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+    boxart: "http://cdn-0.nflximg.com/images/2891/Fracture.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+];
+
+const bestReleases = (releasesArray) => {
+  const selectedReleases = [];
+  releasesArray.forEach((obj) =>
+    obj.rating[0] === 5.0 ? selectedReleases.push(obj.id) : null
+  );
+  return selectedReleases;
+};
+
+// Task 4
+/* С помощью функций map, reduce, вывести url у которого самая большая площадь */
+
+const boxarts = [
+  {
+    width: 200,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture200.jpg",
+  },
+  {
+    width: 150,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture150.jpg",
+  },
+  {
+    width: 300,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture300.jpg",
+  },
+  {
+    width: 425,
+    height: 150,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture425.jpg",
+  },
+];
+
+const getBiggestArea = (boxArray) => {
+  return boxArray.reduce((biggestObj, obj) => {
+    biggestObj =
+      biggestObj.width * biggestObj.height > obj.width * obj.height
+        ? biggestObj
+        : obj;
+    return biggestObj;
+  }).url;
+};
+
+// Task 5
+/* Написать функцию, которая принимает в себя массив из числовых и строчных-числовых значений. Данная ф-ция должна вернуть массив, в которос числовые значения исходного массива стали строчными, а строчные - числовыми */
+
+const convertValues = (valArray) =>
+  valArray.map((value) =>
+    typeof value === "string" ? Number(value) : String(value)
+  );
+
+// Task 6
+/* Написать функцию, которая принимает в себя 2 массива, а возвращает один, состоящий из 2ух, которые пришли в нее */
+
+const joinArrays = (arr1, arr2) => {
+  // Check lenght of each array, add values (do push) to the biggest array to decrease number of iterations for forEach
+  if (arr1.length > arr2.length) {
+    arr2.forEach((val) => arr1.push(val));
+    return arr1;
+  } else {
+    arr1.forEach((val) => arr2.push(val));
+    return arr2;
+  }
+};
```

Comment:
this could be simlifyed: `return [...arr1, ...arr2]` or `return arr1.concat(arr2)`. 
There is a problem in your solution - you mutate intire array

Reply:
fixed